homeassistant/components/rainmachine/util.py

```diff
@@ -35,6 +38,60 @@ class RunStates(StrEnum):
 }
 
 
+@dataclass
+class EntityDomainReplacementStrategy:
+    """Define an entity replacement."""
+
+    old_domain: str
+    old_unique_id: str
+    replacement_entity_id: str
+    breaks_in_ha_version: str
+    remove_old_entity: bool = True
+
+
+@callback
+def async_finish_entity_domain_replacements(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    entity_replacement_strategies: Iterable[EntityDomainReplacementStrategy],
+) -> None:
+    """Remove old entities and create a repairs issue with info on their replacement."""
+    ent_reg = entity_registry.async_get(hass)
+    for strategy in entity_replacement_strategies:
+        try:
+            [registry_entry] = [
+                registry_entry
+                for registry_entry in ent_reg.entities.values()
+                if registry_entry.config_entry_id == entry.entry_id
+                and registry_entry.domain == strategy.old_domain
+                and registry_entry.unique_id == strategy.old_unique_id
+            ]
+        except ValueError:
+            continue
+
+        old_entity_id = registry_entry.entity_id
+        translation_key = "replaced_old_entity"
+
+        async_create_issue(
```

Comment:
We don't create issues just because a platform is set up. We don't know that the user is using this sensor in automations or scripts, so the user may not need to do anything. If we don't know that there is something to do for the user we should not create a repairs issue.

Reply:
Hmm, okay. I fee like I've gotten different feedback on this issue before, but I'm good to change. Would it be reasonable to log a message instead? I don't trust that people are going to just read the documentation, so I would like some sort of proactive warning.

Reply:
I think it's the same case there. We don't know that the user needs to do anything. We shouldn't log a warning in that case.

Reply:
Just realized that it pops back up on every restart so I think it's probably a good idea to remove.

Reply:
Roger. I'll get a PR put together shortly.

Reply:
Thanks 🙏